catalog/transformer/catalog-transformer-csv-common/src/main/java/ddf/catalog/transformer/csv/common/CsvTransformer.java

```diff
@@ -0,0 +1,178 @@
+ * @see Metacard
+ * @see QueryResponseTransformer
+ * @see QueryResponse
+ * @see Attribute
+ */
+public class CsvTransformer {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CsvTransformer.class);
+
+  private static final MimeType CSV_MIME_TYPE;
+
+  static {
+    try {
+      CSV_MIME_TYPE = new MimeType("text/csv");
+    } catch (MimeTypeParseException e) {
+      throw new ExceptionInInitializerError(e);
+    }
+  }
+
+  public static BinaryContent createResponse(final Appendable csv) {
+    InputStream inputStream =
+        new ByteArrayInputStream(csv.toString().getBytes(StandardCharsets.UTF_8));
+    return new BinaryContentImpl(inputStream, CSV_MIME_TYPE);
+  }
+
+  public static Appendable writeMetacardsToCsv(
+      final List<Metacard> metacards,
+      final List<AttributeDescriptor> sortedAttributeDescriptors,
+      final Map<String, String> aliasMap)
+      throws CatalogTransformerException {
+    StringBuilder stringBuilder = new StringBuilder();
+
+    try {
+      CSVPrinter csvPrinter = new CSVPrinter(stringBuilder, CSVFormat.RFC4180);
+      printColumnHeaders(csvPrinter, sortedAttributeDescriptors, aliasMap);
+
+      metacards.forEach(m -> printMetacard(csvPrinter, m, sortedAttributeDescriptors));
+
+      return csvPrinter.getOut();
+    } catch (IOException ioe) {
+      throw new CatalogTransformerException(ioe);
+    }
+  }
+
+  private static void printMetacard(
+      final CSVPrinter csvPrinter,
+      final Metacard metacard,
+      final List<AttributeDescriptor> sortedAttributeDescriptors) {
+
+    Iterator<Serializable> metacardIterator =
+        new MetacardIterator(metacard, sortedAttributeDescriptors);
+
+    printData(csvPrinter, metacardIterator);
+  }
+
+  private static void printColumnHeaders(
+      final CSVPrinter csvPrinter,
+      final List<AttributeDescriptor> sortedAttributeDescriptors,
+      final Map<String, String> aliasMap) {
+    Iterator<String> columnHeaderIterator =
+        new ColumnHeaderIterator(sortedAttributeDescriptors, aliasMap);
+
+    printData(csvPrinter, columnHeaderIterator);
+  }
+
+  private static void printData(final CSVPrinter csvPrinter, final Iterator iterator) {
+    try {
+      csvPrinter.printRecord(() -> iterator);
+    } catch (IOException ioe) {
+      LOGGER.debug("Failed to print the CSV data.", ioe);
+    }
+  }
+
+  public static List<AttributeDescriptor> sortAttributes(
+      final Set<AttributeDescriptor> attributeSet, final List<String> attributeOrder) {
+    CsvAttributeDescriptorComparator attributeComparator =
+        new CsvAttributeDescriptorComparator(attributeOrder);
+
+    return attributeSet.stream().sorted(attributeComparator).collect(toList());
+  }
+
+  /**
+   * Given a list of {@link Metacard}s and a string set of hiddenFields, returns a set of {@link
+   * AttributeDescriptor}s that contains all attributes that exist on the given metacard types
+   * except those that should be excluded.
+   */
```

Comment:
These Javadocs lack important pieces to be complete like `@param` and `@return`. The Javadoc should either be completed or the methods should be self-explanatory enough to allow the Javadocs to be removed. (I think that with a method name change or two where mentioned, the methods in here are pretty self-explanatory and might not require Javadoc.)